backend/server/src/services/matchService.ts

```diff
@@ -193,9 +198,85 @@ export class MatchService {
     if (player1Points >= MAXIMUM_POINTS) {
       match.winner = player1.id;
       match.endTimestamp = new Date();
+      await this.createPlayoffSchedule(match.id, player1.id);
     } else if (player2Points >= MAXIMUM_POINTS) {
       match.winner = player2.id;
       match.endTimestamp = new Date();
+      await this.createPlayoffSchedule(match.id, player2.id);
+    }
+  }
+
+  private async createPlayoffSchedule(
+    matchId: Types.ObjectId,
+    winnerId: Types.ObjectId
+  ): Promise<void> {
+    const tournament = await TournamentModel.findOne({
+      matchSchedule: matchId
+    }).exec();
+    if (tournament?.type !== TournamentType.Playoff) {
+      return;
+    }
+
+    const playedMatches = await MatchModel.find({
+      _id: { $in: tournament.matchSchedule }
+    }).exec();
+
+    const currentMatch = playedMatches.find(
+      (match) => match.id.toString() === matchId.toString()
+    );
+    if (currentMatch === null || currentMatch === undefined) {
+      throw new NotFoundError({
+        message: "Match not found in tournament schedule"
+      });
+    }
+    const currentRound = currentMatch.tournamentRound;
+    const nextRound = currentRound + 1;
+
+    const winners = playedMatches
+      .filter((match) => match.tournamentRound === currentRound && match.winner)
+      .map((match) => match.winner)
+      .filter((winner): winner is Types.ObjectId => winner != null);
```

Comment:
can the winner be null? afaik the winner is either an id or undefined. Or does that matter here?

Reply:
It doesn't matter here. I added the null checker because TS was causing a scene.